src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.smartdashboard.*;
+
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.controls.VelocityVoltage;
+
+public class Shooter extends SubsystemBase {
+
+  private final TalonFX pivotMotor = new TalonFX(0); // TODO: Set to pivot motor device Id
+  private final TalonFX leftShootMotor = new TalonFX(0); // TODO: Set to left motor device Id
+  private final TalonFX rightShootMotor = new TalonFX(0); // TODO: Set to left motor device Id
+  private final TalonFX intakeMotor = new TalonFX(0); // TODO: Set to left motor device Id
+  private double speed = 0;
+  private VelocityVoltage velocity = new VelocityVoltage(speed);
+  private String position = "Default";
+  public boolean canShoot = false;
+
+  public Shooter() {}
+
+  public void shooterInit() {
+    SmartDashboard.putNumber("Shooter Motor Speed", speed);
+    SmartDashboard.putString("Shooter Position", position);
+    leftShootMotor.setInverted(true); // TODO: Change to Right/Left to invert shooting motor
+  }
+
+  public void startMotors(double speed, String position) {
+    this.speed = speed;
+    this.position = position;
+    SmartDashboard.putNumber("Shooter Motor Speed", speed);
+    SmartDashboard.putString("Shooter Position", position);
+    if (position == "Amp") {
+      pivotMotor.setPosition(0.1);  // TODO: Change to fit amp angle
+    } else if (position == "Speaker") {
+      pivotMotor.setPosition(0.05);  // TODO: Change to fit speaker angle
+    }
+    rightShootMotor.setControl(velocity.withVelocity(speed));
+    leftShootMotor.setControl(velocity.withVelocity(speed));
+    canShoot = true;
+  }
+
+  public void shoot(double speed) {
+    if (this.speed != 0 && (this.position == "Default")) { // If shooter is in position and ready to fire
```

Comment:
This is the other reason to not use `canShoot` as your field value since you only want to shoot if multiple conditions are met. The `speed` one is handled based on my previous comment around `motorsAtShootingSpeed`, but the position one isn't. Maybe the `canShoot` method needs to check the position as well and return false if the position isn't at Default? (Note: I assume that's what you meant here, to say `this.position != "Default"`)

Now you have a question on whether you want to have your `shoot` method check `canShoot` or whether you expect consumers to do so. And if `canShoot` is false when `shoot` is called, do you throw an exception? One style here would be to check `canShoot` here and if it's false, return `false` from the `shoot` method (you'd have to make it return boolean instead of `void`). Only once you've spun up your intake motors would `canShoot` be false. But if we're using `motorsAtShootingSpeed`, we wouldn't actually be changing the speed of the motors of our shooter. And `intakeMotor` may not be under this subsystem's control. Hmmm. 🤔 

It almost seems like the actual action of `shooting` shouldn't be handled here because it involves multiple subsystems. Instead, the `Shoot` command would entirely handle moving the intake motor (through the subsystem) instead of calling `shoot` directly on this subsystem. Now we can still do our `canShoot` method on our shooter and we only start our timer in the `Shoot` command if we can.